app/assets/javascripts/discourse/app/lib/array-tools.js

```diff
@@ -0,0 +1,34 @@
+import {
+  beginPropertyChanges,
+  endPropertyChanges,
+} from "@ember/-internals/metal";
+import { TrackedArray } from "@ember-compat/tracked-built-ins";
+
+function _removeValue(array, value) {
+  let loc = array.length || 0;
+  while (--loc >= 0) {
+    if (array[loc] === value) {
+      array.splice(loc, 1);
+    }
+  }
+  return array;
+}
+
+/**
+ * Removes all occurrences of a specified value from an array.
+ *
+ * @param {Array} array - The array from which the value needs to be removed.
+ * @param {*} value - The value to be removed from the array.
+ * @return {Array} The modified array with the specified value removed.
+ */
+export function removeObject(array, value) {
+  if (array instanceof TrackedArray) {
+    beginPropertyChanges();
```

Comment:
Using Ember internals to batch updates in case more than one item is removed from the tracked array

Reply:
In this case, probably not. Odds are that most case will have only one hit anyway.

But the Ember source code does this under the hood for the `removeObjects` function, that will be handled in another PR. In that case, I'd advocate more firmly to use this strategy.